### DISCUSSION TITLE
Behringer FCB1010

### DISCUSSION THREAD

**Discussion by Wapata** [alf45tar/PedalinoMini#440] [General]
Hi there !
The project look very cool, and it seem the configuration is well made !
Then... I have a Behringer FCB1010, with original firmware, the hardware is rock solid but the software... Bad.

Do you think it's a bad/good idea to empty the Behringer FCB1010 and replace everything inside to run it as PedalinoMini ?
Is see a lot of pro: midi over BT, low power conso so it can be battery powered (with the space inside it surely can power other devices to), web configuration...
I'm not able to see if there is cons... Except that I'm not sure if the analog pedals can work easily (it's an optical light sensor) out of the box.

Open discussion, let's see what YOU think ;-)

**Comment by rigr** [comment]
Looks like a lot of work required to modify the FCB1010 so iot can use a ESP32 - do you know https://el34world.com/Misc/Music/SoundPage1.htm ?

**Reply by Wapata** [reply]
Thanks for your answer. Yes I think I've seen this page before, and some other about retrofitting the FCB1010, passing the power thru the midi jack etc... And then the Uno ship and variations of it. In the end, the Uno ship costs are not low, and the editor software neither (and not multi platform).
In the end, it's been two years since I know the product and didn't buy it, still looking for an other solution.

**Comment by alf45tar** [comment]
I think it is no so complex and it is possible to create a mod that it is reversible because all the hardware is connected to the motherboard with connectors.

**Reply by Wapata** [reply]
Thanks for the answer too !
I'm a little bit lost in one thing about PedalinoMini: is it a "finished" project or is it still forked into two versions ?
I have to check if I have the minimal hardware to play a little with the web interface and see if I like it, then sketch the wiring if everything seem fun to do !

**Reply by alf45tar** [reply]
PedalinoMini is a never finished project. Software is updated regularly and sometime new hardware boards are added too. Sponsor version includes new features and bug corrections to be released to public version after some months.

**Comment by Wapata** [comment]
So far, it look it should work !
![IMG_20230624_125516_405](https://github.com/alf45tar/PedalinoMini/assets/15179434/93b268a7-0470-4792-a53d-b1ada3bb2f59)

**Comment by Wapata** [comment]
Hi !
So, as an update:
![IMG_20230917_181217_336](https://github.com/alf45tar/PedalinoMini/assets/15179434/bbc39e9c-84ca-4620-8049-183f8d78bd44)
Legacy midi and all buttons and pedal are working great.
Look quite good in the outside, it's a mess inside.
I don't recommend to do it like this except if you know and can re-do PCBs to hold everything in a good position. I don't know how to do it yet, so my LEDs do not stay in place.
The original red screen make the bank 3 invisible... I have to remake the whole upper screen to have a transparent one, with new labels and new led placement... as well as a new PCB under to hold the ESP board.
Then in the back, make adapters to replace the big power socket by to USB connectors (programing port and usb-midi port).
And maybe add a battery.
The jack in the back are now usable by PedalinoMini :-)

I have to look for having midi over Ethernet rj45 port, should be more stable than wifi, and the Arduino than create the midi-usb should hopefully handle that.

So.
Thanks a lot for this project ! Without a woman and a kid, it surely can be a one week project for anyone !

**Reply by mike76700** [reply]
hello
how did you configure you're pedal? i do the same but operation is not great.

**Reply by Wapata** [reply]
I've made tow analog, two ladder, and two empty jacks.
If I find how to share my google image album with you in private i can give you a lot of my data (but i can't share it to the www) to compare.
What feel wrong from your side ?

**Comment by morpheus1306** [comment]
Hi guys, I went down that route, too: 
It's german thread but maybe it is usefull: https://www.musiker-board.de/threads/fcb1010-mit-modernem-mikrocontroller-bestuecken.756515/

https://github.com/T-vK/PedalinoMini 

I noticed, he did not use LEDs for the expression pedals. So I need to set the number of LEDs to 12.

Right now I am just testing with a some switches and pots. 
I already ripped apart the FCB1010 , because I converted it from 230V to 5V USB and used an internal USB hub and a DIN-MIDI2USB cable.
But somehow I could not get smooth expression pedal signals and I decided to go the PedalinoMini route. 
Amazing stuff! Great work!

I got one question about the configuration: With the FCB1010 UnO Chip I had 10 banks with 10 presets and it was possible to assign different CC# to the Exp A and Exp B pedals per preset. There are 10 preset, 2 buttons for bank + and bank -, and 2 expression pedals.

How do I set this up with the PedalinoMini firmware?
I have 14 controls, 1-10 presets select foot switches, 2 bank+/- switches and 2 analog exp pedals.

But how could I change the behavior of the exp pedals when pressing e.g. control 1 ?

Thanks for short comment!

**Reply by alf45tar** [reply]
For an analog pedal leave Send blank to activate the universal mode. In universal mode an analog pedal can be used to repeat the last MIDI message. A typical usage is to modify the value of the last CONTROL CHANGE.

**Reply by morpheus1306** [reply]
Hm, ok. But how could that mimic the behavio I would like to have?
So I need two actions like one send Program Change and one e.g. send CC02 ? And the Pedal will then send out CC02s on movements?

What are these slots?

**Reply by morpheus1306** [reply]
I though I could link different analog pedal actions to the selected "presets" so when hitting control 2, with selected slot 2 its picking the slot 2 action from the analog pedal action. But obviously that not the case.
